trade-finance-manager-ui/server/controllers/helpers/amendments.helper.show-amendment-button.test.js

```diff
@@ -0,0 +1,90 @@
+import { TEAM_IDS, DEAL_STATUS } from '@ukef/dtfs2-common';
+import { showAmendmentButton } from './amendments.helper';
+import { AmendmentButtonDealBuilder } from '../../test-mocks/show-amendments-button-deal-builder';
+import { DEAL } from '../../constants';
+
+describe('amendments helper', () => {
+  describe('showAmendmentButton()', () => {
+    const defaultUserTeams = [TEAM_IDS.PIM];
+
+    it.each(getSuccessTestCases())('should return true when $description', ({ deal, userTeams }) => {
+      const result = showAmendmentButton(deal, userTeams);
+      expect(result).toEqual(true);
+    });
+
+    it.each(getFailureTestCases())('should return false when $description', ({ deal, userTeams }) => {
+      const result = showAmendmentButton(deal, userTeams);
+      expect(result).toEqual(false);
+    });
+
+    function getSuccessTestCases() {
```

Comment:
I've reworked these tests to explicity increase the coverage while greatly reducing LoC.

These have to be functions to be under the `meat and gravy` of the tests